src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -26,8 +26,59 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
 void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
 {
-    NamedIntrinsic intrinsicID = node->gtHWIntrinsicId;
-    InstructionSet isa         = compiler->isaOfHWIntrinsic(intrinsicID);
+    NamedIntrinsic      intrinsicID = node->gtHWIntrinsicId;
+    InstructionSet      isa         = compiler->isaOfHWIntrinsic(intrinsicID);
+    HWIntrinsicCategory category    = compiler->categoryOfHWIntrinsic(intrinsicID);
+
+    if (category == HW_Category_SimpleSIMD)
```

Comment:
nit: We should probably have a `switch(category)` and `genHWIntrinsicSimpleSIMD` function, etc

Reply:
Switch may not work with mask bit now, but we can consider to define several enum to use switch later.